### ISSUE TITLE
Add PunchDown connectors 

### ISSUE THREAD

**Issue by minkione** [exploitagency/ESP-RFID-Thief#1]
Would be way more effective to replace the green connectors with the punchdown ones.
Then it would be easier to install them on existing readers during an engagement.

http://www.ebay.ie/itm/4x-009175003001006-Connector-IDC-transition-9175-punch-down-block-ways3-AVX-/201955111756

https://www.tme.eu/en/details/009175003001006/led-connectors/avx/

**Comment by exploitagency** [comment]
It looks as easy as soldering them in place of the screw terminal block...

From the eBay ad:
> Contacts pitch 2.5mm

So that is up to the end user assembling the board.

Not everyone has a punch down tool or wants to deface wiring, it strips the sheath when you punch it down.

**Comment by exploitagency** [comment]
I like the screw terminal idea better, not everyone wants to deface their wiring, punch down means you will strip the sheath on the wiring.  This device is designed primarily to create a standalone reader like with the "Tastic RFID Thief", punch down could actually come loose easier in this situation.  Dropping these units into existing RFID reader implementations is unlikely in a legal pen testing situation.

Its possible they make the punch down blocks in the same pin spacing and through hole so no design change may be needed if that is the case and the end user can assemble it that way.

**Comment by minkione** [comment]
Actually weaponizing external readers makes a lot of sense, since u place it and leave.
Wait couple of days that people will enter the building... come back and collect all the IDs you want.
https://twitter.com/LucaBongiorni/status/908935763022229504
https://redteamtools.com/espkey
I was thinking to create the cheaper version, but you saved my time 😃 👍 

P.S. Since I am adopting both scenarios (i.e. portable maxiprox longrange and weaponizing customer's readers), by my experience, I can say that is way faster the latter option.

**Comment by exploitagency** [comment]
I know punch down will be faster.
I will consider it.

I actually released my PoC months before the ESPKey came out, I actually wonder if they took my idea.  That is why this hardware is licensed, if someone makes modifications they need to give credit.

**Comment by minkione** [comment]
Not sure who copied who. 
Before ESPkey, there was also BLEkey https://github.com/linklayer/BLEKey
And before it there was the Tastic RFID (that IIRC didn't have any wireless connection, but SD storage).

**Comment by exploitagency** [comment]
I released my first PoC Sep 11, 2016 and didn't see ESPKey until this year around March/April.
Somehow I missed out on news of BLEKey.
My version is a direct port of Tastic RFID Thief to the ESP8266

**Comment by minkione** [comment]
Well, as customer that bought that ESPKey, I was expecting them to release the sources as promised... but so far nothing... Quite disappointing, considering the insane price for an ESP-12 and a bunch of few more components.
Here the devguy of ESPkey https://twitter.com/octosavvi/status/919688934774398976

**Comment by exploitagency** [comment]
I am open from the beginning and make it clear that my software is a port of Tastic with additional features.

One reason I am suspicious is this guys source isn't released.  But in his defense that is one way to keep his price sky high.  Also I don't think he has a programming header.  Which by the way I suggest the user use pogo pins or clips to program and not actually solder on a header to the pads.  I support OTA HTTP updates so it will be easy after initial programming.

**Comment by minkione** [comment]
Yeah your programming header is way better for easy debugging/flashing/unbricking!
Overall your solution looks nice and way cheap!
I will share it on Twitter!
Think about adding a second gerber files for the pushdown connectors though 😄

**Comment by exploitagency** [comment]
No problem I will do it!
The punch down style will be an untested footprint though until I have the modified pcb and the proper connector in hand.

**Comment by minkione** [comment]
Add me on Twitter, I need to DM you

**Comment by exploitagency** [comment]
done

**Comment by exploitagency** [comment]
I originally thought if someone wanted to field deploy into an existing reader installation they could use splice connectors like these.
https://ae01.alicdn.com/kf/HTB16RsFMpXXXXbIXVXXq6xXFXXXS/10pcs-lot-L1-Blue-T-Type-Quick-font-b-Splice-b-font-font-b-Crimp-b.jpg

**Comment by exploitagency** [comment]
Reference the BOM for the proper part #'s.

https://github.com/exploitagency/ESP-RFID-Thief/tree/master/Board/UNTESTED%20-%20IDC%20Punch%20Down%20Block%20Version
NOTE: Once this new design is tested the link will be dead as folder will be renamed.

I used what I found on Mouser... unsure which gauge IDC to use, larger gauge, smaller gauge or something in the middle.  I need to find what is the most reliable across several wire options.  Perhaps smaller gauge connector will stretch and fit larger gauge or larger gauge connector won't skin the smaller gauge wire, or perhaps I should try something in the middle.  I have a choice of 18,20,22,24 connectors.  It has to support the most commonly found wire sizes, so narrowing down the actual part number for the connector will take some actual testing(I have various AWG wire) as well as waiting on parts to arrive in the first place.
